pkg/virt-handler/device-manager/generic_device.go

```diff
@@ -0,0 +1,262 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2018 Red Hat, Inc.
+ *
+ */
+
+package device_manager
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"path"
+	"strconv"
+	"time"
+	"path/filepath"
+
+	"golang.org/x/net/context"
+
+	"google.golang.org/grpc"
+	pluginapi "k8s.io/kubernetes/pkg/kubelet/apis/deviceplugin/v1alpha"
```

Comment:
That should be the only dependecy you need: `k8s.io/kubernetes/pkg/kubelet/apis/deviceplugin/v1beta1`. Also make sure that it aligns with the version which we use for client-go, currently `kubernetes-1.9.7`.

Reply:
Something like

```diff
diff --git a/glide.yaml b/glide.yaml
index 784b28df..61ff8ee0 100644
--- a/glide.yaml
+++ b/glide.yaml
@@ -52,6 +52,10 @@ import:
   version: ^0.0.2
   subpackages:
   - cobra
+- package: k8s.io/kubernetes
+  version: kubernetes-1.9.7
+  subpackages:
+  - /pkg/kubelet/apis/deviceplugin/v1beta1
 - package: k8s.io/apiextensions-apiserver
   version: kubernetes-1.9.7
 - package: github.com/ant31/crd-validation
```

should do it

Reply:
Edit: If you use `v1.10.0` you can use `v1beta`. For `v1.9.7` you will have to use `v1alpha1`.

Reply:
Actually we should merge this after Kubernetes 1.10.X is used as a provider. Requiring a feature-gate to use an alpha feature (in 1.9.X) is suboptimal.

Reply:
This PR is probably gated on: https://github.com/kubevirt/kubevirt/pull/1145

Reply:
We have k8s 1.10.3 as default for some time already